Audiobook Player/PlayerManager.swift

```diff
@@ -157,8 +158,18 @@ class PlayerManager: NSObject {
                     MPMediaItemPropertyArtwork: mediaArtwork
                 ]
                 
-                //get stored value for current time of book
-                let currentTime = UserDefaults.standard.integer(forKey: self.identifier)
+                //get stored value for current time of book in seconds
+                var lastPlayedPositionInSeconds = self.defaults.integer(forKey: self.identifier)
+                
+                
+                // If smartRewind is enabled and time since last play was 10 minutes (599s), rewind audiobook by 30 seconds or to start.
+                if let lastPlayTime:Date = self.defaults.object(forKey: UserDefaultsConstants.lastPauseTime+"_\(self.identifier)") as? Date, self.defaults.bool(forKey: UserDefaultsConstants.smartRewindEnabled)  {
+                    if Date().timeIntervalSince(lastPlayTime) > 599 {
+                        lastPlayedPositionInSeconds = max(lastPlayedPositionInSeconds - 30, 0)
```

Comment:
just like we store the `lastPauseTime` on the pause event, I think we should clear the time once the smart rewind is triggered once when loading a book, otherwise we would risk having a side effect of triggering more than once for the same recorded time. What do you think 🤔?

Reply:
Hmmm that's probably a good idea. I can't help but think there's probably a cleaner way I can implement the smart rewind check that would get around this issue without having to null out the user default every time, but I can't think of it! I'll add this in!

Reply:
👌 we can figure it out as we move along with the project :D